crates/solidity/outputs/cargo/crate/src/backend/binder/mod.rs

```diff
@@ -161,19 +161,26 @@ impl Binder {
     }
 
     pub(crate) fn insert_scope(&mut self, scope: Scope) -> ScopeId {
-        let node_id = scope.node_id();
         let scope_id = ScopeId(self.scopes.len());
-        if self.scopes_by_node_id.contains_key(&node_id) {
-            unreachable!("attempt to insert duplicate file scope for node {node_id:?}");
-        }
-        if let Scope::File(ref file_scope) = scope {
+
+        if let Scope::File(file_scope) = &scope {
             let file_id = &file_scope.file_id;
             if self.scopes_by_file_id.contains_key(file_id) {
                 unreachable!("attempt to insert duplicate file scope for {file_id}");
             }
             self.scopes_by_file_id.insert(file_id.clone(), scope_id);
         }
-        self.scopes_by_node_id.insert(node_id, scope_id);
+
+        if let Scope::Parameters(_) = &scope {
+            // parameters scope don't have an associated node ID
+        } else {
+            let node_id = scope.node_id();
+            if self.scopes_by_node_id.contains_key(&node_id) {
```

Comment:
here + above: This will search twice. I suggest reusing the return value of `insert()` instead:

```rust
if map.insert(key, value).is_some() {
  unreachable!("attempt to insert duplicate....");
}
```